client/allocrunner/consul_grpc_sock_hook.go

```diff
@@ -184,55 +185,78 @@ func (p *grpcSocketProxy) run(alloc *structs.Allocation) error {
 	default:
 	}
 
-	// make sure either grpc or http consul address has been configured
-	if p.config.GRPCAddr == "" && p.config.Addr == "" {
-		return errors.New("consul address must be set on nomad client")
+	clusterNames := set.New[string](0)
+	tg := alloc.Job.LookupTaskGroup(alloc.TaskGroup)
+	for _, s := range tg.Services {
+		clusterNames.Insert(s.GetConsulClusterName(tg))
 	}
 
-	destAddr := p.config.GRPCAddr
-	if destAddr == "" {
-		// No GRPCAddr defined. Use Addr but replace port with the gRPC
-		// default of 8502.
-		host, _, err := net.SplitHostPort(p.config.Addr)
-		if err != nil {
-			return fmt.Errorf("error parsing Consul address %q: %v",
-				p.config.Addr, err)
+	var retErr error
+
+	clusterNames.ForEach(func(clusterName string) bool {
+		config := p.configs[clusterName]
+
+		// make sure either grpc or http consul address has been configured
+		if config.GRPCAddr == "" && config.Addr == "" {
+			retErr = errors.New("consul address must be set on nomad client")
```

Comment:
```suggestion
			retErr = fmt.Errorf("consul address for cluster %s must be set on nomad client", clusterName)
```

And would it make sense to use a multi-error here so we validate all possible errors instead of one by one?

Reply:
Done, but in the outermost function after making the map of proxies as suggested.